framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+
+  bundleTracker.Open();
+  bundleTracker.Close();
+  
+  EXPECT_EQ(-1, bundleTracker.GetTrackingCount())
+    << "Tracking count of unopened BundleTracker was not -1";
+}
+
+TEST_F(BundleTrackerMethodTest, RemoveWorks)
+{
+  // Given a BundleTracker tracking a bundle
+  BundleTracker<> bundleTracker(context, all_states);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  auto bundlesTracked = bundleTracker.GetBundles();
+  bool bundleTracked =
+    std::find(bundlesTracked.begin(), bundlesTracked.end(), bundle) !=
+    bundlesTracked.end();
+  ASSERT_TRUE(bundleTracked) << "BundleTracker failed to track a bundle";
+
+  // When Remove is called
+  bundleTracker.Remove(bundle);
+
+  // Then the bundle should not be tracked
+  bundlesTracked = bundleTracker.GetBundles();
+  bool bundleStillTracked =
+    std::find(bundlesTracked.begin(), bundlesTracked.end(), bundle) !=
+    bundlesTracked.end();
+  EXPECT_FALSE(bundleStillTracked)
+    << "Calling Remove() didn't removed tracked bundle from being tracked";
+
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, BundleRemovedByRemoveTrackable)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle testBundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+
+  bundleTracker.Remove(testBundle);
+  testBundle.Start();
+
+  bool testBundleTracked = bundleTracker.GetObject(testBundle) == testBundle;
+  EXPECT_TRUE(testBundleTracked)
+    << "Remove() was called on a tracked bundle, then later entered a tracked "
+       "state. The bundle should be tracked but wasn't";
+}
+
+TEST_F(BundleTrackerMethodTest, DefaultAddingBundleReturnsCorrectValue)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  Bundle bundle = Bundle();
+  EXPECT_EQ(bundle, bundleTracker.AddingBundle(bundle, BundleEvent()))
+    << "Default AddingBundle should return the inputted bundle";
+}
+
+TEST_F(BundleTrackerMethodTest, RemoveUntrackedBundleDoesNothing)
+{
+  auto stateMask =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_ACTIVE);
+  BundleTracker<> bundleTracker(context, stateMask);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+
+  int preCount = bundleTracker.GetTrackingCount();
+  ASSERT_NO_THROW(bundleTracker.Remove(bundle))
+    << "Manually removing untracked Bundle should not throw";
+  int postCount = bundleTracker.GetTrackingCount();
+  EXPECT_EQ(preCount, postCount)
+    << "BundleTracker should not change the tracking count when untracked "
+       "Bundle is manually removed";
+
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, ReopeningTrackerWorks)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done.